### ISSUE TITLE
NETSDK1061 version mismatch error after update

### ISSUE THREAD

**Issue by sky-lake** [OmniSharp/omnisharp-vscode#3970]
I recently upgraded my Ubuntu packages (I saw some dotnet packages were updated during this), and now when I start VS Code I have a OmniSharp error:

```
[fail]: OmniSharp.MSBuild.ProjectLoader
        NETSDK1061: The project was restored using Microsoft.NETCore.App version 1.0.5, but with current settings, version 3.0.0 would be used instead. To resolve this issue, make sure the same settings are used for restore and for subsequent operations such as build or publish. Typically this issue can occur if the RuntimeIdentifier property is set during build or publish but not during restore. For more information, see https://aka.ms/dotnet-runtime-patch-selection.
```

I can still build & run normally, both from IDE and command line.

I have also tried a restore from command line, and it did not seem to fix the issue.

I don't think I was having this issue before updating the system packages.

## Environment data
```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  2.1.809 [/usr/share/dotnet/sdk]
  3.1.401 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

VS Code version:  1.47.3
C# Extension version: 1.22.1

## OmniSharp log

```
Starting OmniSharp server at 8/11/2020, 7:22:27 PM
    Target: /project/hatch

OmniSharp server started.
    Path: /home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/run
    PID: 16818

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on ubuntu 18.4 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 16.4 - "/home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin/MSBuild.dll'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.4 - "/home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin
            CscToolPath = /home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /home/sky/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/project/hatch'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/project/hatch'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/project/hatch/hatch.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/project/hatch'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /project/hatch/hatch.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/project/hatch' on host 16060.
[fail]: OmniSharp.MSBuild.ProjectLoader
        NETSDK1061: The project was restored using Microsoft.NETCore.App version 1.0.5, but with current settings, version 3.0.0 would be used instead. To resolve this issue, make sure the same settings are used for restore and for subsequent operations such as build or publish. Typically this issue can occur if the RuntimeIdentifier property is set during build or publish but not during restore. For more information, see https://aka.ms/dotnet-runtime-patch-selection.
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/project/hatch/hatch.csproj'.
/project/hatch/hatch.csproj
/project/hatch/hatch.csproj(0,0): Error: NETSDK1061: The project was restored using Microsoft.NETCore.App version 1.0.5, but with current settings, version 3.0.0 would be used instead. To resolve this issue, make sure the same settings are used for restore and for subsequent operations such as build or publish. Typically this issue can occur if the RuntimeIdentifier property is set during build or publish but not during restore. For more information, see https://aka.ms/dotnet-runtime-patch-selection.

[fail]: OmniSharp.MSBuild.ProjectManager
        Attempted to update project that is not loaded: /project/hatch/hatch.csproj
```

**Comment by JoeRobich** [comment]
@sky-lake hmm. It sounds like the project.assets.json file may be stale. If you run dotnet restore against your project, does it load properly the next time it is opened?

**Comment by sky-lake** [comment]
Hello!

I just tried deleting the whole obj folder of both my exe and library project.  Then I tried executing `dotnet restore` for both in terminal.  When I load up VS code, I still get the error.  I don't get the error if from terminal I do `dotnet build` or `dotnet publish`

**Comment by JoeRobich** [comment]
@sky-lake It seems you are running into some version of https://github.com/dotnet/sdk/issues/2682

Can you share your project file?

**Comment by sky-lake** [comment]
Interesting!  Here is my exe project file:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
   <DefineConstants Condition="'$(SET_PRIVBUILD)'=='true'">$(DefineConstants);PRIVBUILD</DefineConstants>
   <DefineConstants Condition="'$(SET_LOCALGAMETEXTPRIVATE)'=='true'">$(DefineConstants);LOCALGAMETEXTPRIVATE</DefineConstants>
  </PropertyGroup>

  <ItemGroup>
    <Compile Update="**\*.xaml.cs">
      <DependentUpon>%(Filename)</DependentUpon>
    </Compile>
    <AvaloniaResource Include="**\*.xaml">
      <SubType>Designer</SubType>
    </AvaloniaResource>
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="0.9.0" />
    <PackageReference Include="Avalonia.Desktop" Version="0.9.0" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\core\core.csproj" />
  </ItemGroup>
</Project>
```

Here is my lib project file (core.csproj):

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <OutputType>Library</OutputType>
    <DefineConstants Condition="'$(SET_PRIVBUILD)'=='true'">$(DefineConstants);PRIVBUILD</DefineConstants>
    <DefineConstants Condition="'$(SET_LOCALGAMETEXTPRIVATE)'=='true'">$(DefineConstants);LOCALGAMETEXTPRIVATE</DefineConstants>
  </PropertyGroup>

   <!-- Embedded method:
  <ItemGroup>
   <Content Remove="GameText.xml" />
   <Content Remove="GameTextPrivate.xml" />
  </ItemGroup>
  <ItemGroup>
   <EmbeddedResource Include="GameText.xml" />
   <EmbeddedResource Include="GameTextPrivate.xml" />
  </ItemGroup>
  -->

  <ItemGroup>
   <Content Include="GameText.xml" />
   <Content Include="GameTextPrivate.xml" />  <!-- TODO: Don't include on public build, if it matters.-->
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>

</Project>
```

This is my first dotnet core project, so please forgive any newbie mistakes 🙏

**Comment by michellamin** [comment]
Had the same issue. Aparently the sdk was working just fine, since a could build my projects via CLI while vscode couldn't resolve my dependencies.

Had to rollback the sdk version e set it in the project's gloabl.json file.

1 - Download past release https://dotnet.microsoft.com/download/dotnet-core/3.1
2 - Move the contents from /dotnet/sdk/3.1.107/ to /usr/share/dotnet/sdk/ (Ubuntu)
3 - Set previous version o global.json file:
`{
  "sdk": {
    "version": "3.1.107"
  }
}`

**Comment by pyrrho** [comment]
I think I'm having an issue similar to @michellamin and @sky-lake. I upgraded from 3.1.300 to 3.1.401 and vscode began showing numerous errors in code that don't prevent the CLI from correctly building and testing the project. (The errors are somewhat nonsensical; `The type 'Enum' is defined in an assembly that is not referenced.`) Looking through the logs, I saw the `Microsoft.Build.Exceptions.InvalidProjectFileException: The SDK 'Microsoft.NET.Sdk' specified could not be found` error, which is how I wound up here.

Downgrading to 3.1.302 seems to "fix" the omnisharp extension, in that the spurious errors are no longer generated.

Here's the interesting part. The `Microsoft.Build.Exceptions.InvalidProjectFileException:` error is still emitted in vscode's OmniSharp logs when using 3.1.302.

Perhaps Microsoft has been playing with its package layout, moved the SDK a few version back, and just recently moved the `System`, `System.Void`, etc. packages into that misplaced SDK?

**Comment by JoeRobich** [comment]
@pyrrho Issues with using the latest .NET Core SDK with our included Mono should be resolved in this pre-release - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.23.0

**Comment by sky-lake** [comment]
Can confirm the omnisharp that VS Code pulls via 'latest' appears to have fixed this issue as of yesterday
`/home/sky/.vscode/extensions/ms-dotnettools.csharp-1.23.1/.omnisharp/1.37.1-beta.2/run`

**Comment by jmarolf** [comment]
fixed in 1.23.1
